utility_analysis_new/combiners.py

```diff
@@ -215,3 +215,130 @@ def metrics_names(self) -> List[str]:
+            ps_accumulator)
+
+
+class UtilityAnalysisSumCombiner(pipeline_dp.Combiner):
+    """A combiner for utility analysis sums."""
+    AccumulatorType = UtilityAnalysisSumAccumulator
+
+    def __init__(self, params: pipeline_dp.combiners.CombinerParams,
+                 is_public_partitions: bool):
+        self._params = params
+        self._is_public_partitions = is_public_partitions
+
+    def create_accumulator(self, data: Tuple[Sized, int]) -> AccumulatorType:
+        """Create an accumulator for data.
+
+        Args:
+            data is a Tuple containing; 1) a list of the user's contributions for a single partition, and 2) the total
+            number of partitions a user contributed to.
+
+        Returns:
+            An accumulator with the sum of contributions and the contribution error.
+        """
+        if not data or not len(data[0]):
+            return UtilityAnalysisSumAccumulator(0, 0, 0, 0, None)
+        values, n_partitions = data
+        max_partitions = self._params.aggregate_params.max_partitions_contributed
+        prob_keep_partition = min(1, max_partitions /
+                                  n_partitions) if n_partitions > 0 else 0
+        partition_sum = np.sum(values)
+        clipped_sum = np.clip(
+            partition_sum, self._params.aggregate_params.min_sum_per_partition,
+            self._params.aggregate_params.max_sum_per_partition)
+        per_partition_contribution_error = partition_sum - clipped_sum
+        count = len(values)
+        max_per_partition = self._params.aggregate_params.max_contributions_per_partition
+        per_partition_contribution = min(max_per_partition, count)
+        expected_cross_partition_error = -per_partition_contribution * (
+            1 - prob_keep_partition)
+        var_cross_partition_error = per_partition_contribution**2 * prob_keep_partition * (
+            1 - prob_keep_partition)
+
+        ps_accumulator = None
+        if not self._is_public_partitions:
+            ps_accumulator = PartitionSelectionAccumulator(
+                probabilities=[prob_keep_partition])
+
+        return UtilityAnalysisSumAccumulator(partition_sum,
+                                             per_partition_contribution_error,
+                                             expected_cross_partition_error,
+                                             var_cross_partition_error,
+                                             ps_accumulator)
+
+    def merge_accumulators(self, acc1: AccumulatorType, acc2: AccumulatorType):
+        """Merge two accumulators together additively."""
+        return acc1 + acc2
+
+    def compute_metrics(self,
+                        acc: AccumulatorType) -> SumUtilityAnalysisMetrics:
+        """Compute metrics based on the accumulator properties.
+
+        Args:
+            acc: the accumulator to compute from.
+
+        Returns:
+            A SumUtilityAnalysisMetrics object with computed metrics.
```

Comment:
no need in the type of output, it's already indicated in definition, after removing that, it looks like 1 line comment is enough (Args and Returns follow from definition)
Nit: leave only 1 line comment